### PR TITLE
Minimal Docker runtime

### PR DESCRIPTION
**Pull request by Wirone**
Allow contributors work on the code without actual PHP8 on their computers.

[Initial discussion](https://github.com/rectorphp/rector/discussions/6411#discussioncomment-1469896).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,6 +2,19 @@
...
+### Preparing local environment
+
+If you have PHP8 and Composer installed locally you can use it straight away
+if `composer check-platform-reqs` results with success.
+
+Alternatively you can use Docker runtime. All you need to do is wrap every command
+with `docker-compose run php`, so commands will be executed inside Docker container.
+
+For example `docker-compose run php composer install` will download PHP dependencies,
+so you can start using all scripts and work with the code.
```
Comment (TomasVotruba) [inline CONTRIBUTING.md]: We're using native line breaks by convention in MD files.

```suggestion
If you have PHP8 and Composer installed locally you can use it straight away if `composer check-platform-reqs` results with success.

Alternatively you can use Docker runtime. All you need to do is wrap every command with `docker-compose run php`, so commands will be executed inside Docker container.

For example `docker-compose run php composer install` will download PHP dependencies, so you can start using all scripts and work with the code.
```

Comment (TomasVotruba) [inline CONTRIBUTING.md]: We're using native line breaks by convention in MD files.

```suggestion
If you have PHP8 and Composer installed locally you can use it straight away if `composer check-platform-reqs` results with success.

Alternatively you can use Docker runtime. All you need to do is wrap every command with `docker-compose run php`, so commands will be executed inside Docker container.

For example `docker-compose run php composer install` will download PHP dependencies, so you can start using all scripts and work with the code.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,6 +2,19 @@
...
+
+### Preparing pull request
```
Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
### Preparing Pull Request
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,6 +2,19 @@
...
+
 There 3 rules will highly increase changes to get your PR merged:
```
Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
There 3 rules will highly increase chance to get your PR merged:
```

Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
There 3 rules will highly increase chance to get your PR merged:
```

Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
There 3 rules will highly increase chance to get your PR merged:
```

Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
There 3 rules will highly increase chance to get your PR merged:
```

Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
There 3 rules will highly increase chance to get your PR merged:
```

Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
There 3 rules will highly increase chance to get your PR merged:
```

**Comment by Wirone** [comment]
Wow, Github went crazy with those suggestions 🤯 😁 Should I apply them or you will clean it up @TomasVotruba ?

**Comment by TomasVotruba** [comment]
Yea, this train has very poor wifi :D 

I'll handle it :+1:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,7 +2,17 @@
 
 Contributions here are more than welcomed! You can contribute to [rector-src](https://github.com/rectorphp/rector-src) repository.
 
-There 3 rules will highly increase changes to get your PR merged:
+### Preparing local environment
```
Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
### Preparing Local Environment
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -2,7 +2,17 @@
...
+
+If you have PHP8 and Composer installed locally you can use it straight away if `composer check-platform-reqs` results with success.
```
Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
If you have PHP 8 and Composer installed locally you can use it straight away. You can validate your environment with:

   ```bash
   composer check-platform-reqs
   ```
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -2,7 +2,20 @@
 
 Contributions here are more than welcomed! You can contribute to [rector-src](https://github.com/rectorphp/rector-src) repository.
 
-There 3 rules will highly increase changes to get your PR merged:
+### Preparing Local Environment
+
+If you have PHP 8 and Composer installed locally you can use it straight away. You can validate your environment with:
+
+   ```bash
+   composer check-platform-reqs
```
Comment (TomasVotruba) [inline CONTRIBUTING.md]:
```suggestion
```bash
composer check-platform-reqs
```
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -2,7 +2,20 @@
 
 Contributions here are more than welcomed! You can contribute to [rector-src](https://github.com/rectorphp/rector-src) repository.
 
-There 3 rules will highly increase changes to get your PR merged:
+### Preparing Local Environment
+
+If you have PHP 8 and Composer installed locally you can use it straight away. You can validate your environment with:
+
+```bash
+composer check-platform-reqs
```
Comment (TomasVotruba) [inline CONTRIBUTING.md:10]:
```suggestion
```bash
composer check-platform-reqs
```
```

**Comment by TomasVotruba** [comment]
Hm, so online editing does not support code snippets :D 

I'll edit it locally, so code snippets are copy-paste able. `Inlined code` is hard to mark with cursor and GitHub Markdown shows copy-paste icon only for:

```bash
snippets
```

**Pull request merged**